events/janus_gelfevh.c

```diff
@@ -0,0 +1,476 @@
+int janus_gelfevh_init(const char *config_path);
+void janus_gelfevh_destroy(void);
+int janus_gelfevh_get_api_compatibility(void);
+int janus_gelfevh_get_version(void);
+const char *janus_gelfevh_get_version_string(void);
+const char *janus_gelfevh_get_description(void);
+const char *janus_gelfevh_get_name(void);
+const char *janus_gelfevh_get_author(void);
+const char *janus_gelfevh_get_package(void);
+void janus_gelfevh_incoming_event(json_t *event);
+json_t *janus_gelfevh_handle_request(json_t *request);
+
+/* Event handler setup */
+static janus_eventhandler janus_gelfevh =
+	JANUS_EVENTHANDLER_INIT (
+		.init = janus_gelfevh_init,
+		.destroy = janus_gelfevh_destroy,
+
+		.get_api_compatibility = janus_gelfevh_get_api_compatibility,
+		.get_version = janus_gelfevh_get_version,
+		.get_version_string = janus_gelfevh_get_version_string,
+		.get_description = janus_gelfevh_get_description,
+		.get_name = janus_gelfevh_get_name,
+		.get_author = janus_gelfevh_get_author,
+		.get_package = janus_gelfevh_get_package,
+
+		.incoming_event = janus_gelfevh_incoming_event,
+		.handle_request = janus_gelfevh_handle_request,
+
+		.events_mask = JANUS_EVENT_TYPE_NONE
+	);
+
+/* Plugin creator */
+janus_eventhandler *create(void) {
+	JANUS_LOG(LOG_VERB, "%s created!\n", JANUS_GELFEVH_NAME);
+	return &janus_gelfevh;
+}
+
+
+/* Useful stuff */
+static volatile gint initialized = 0, stopping = 0;
+static GThread *handler_thread;
+static void *janus_gelfevh_handler(void *data);
+static janus_mutex evh_mutex;
+
+/* JSON serialization options */
+static size_t json_format = JSON_INDENT(3) | JSON_PRESERVE_ORDER;
+
+/* Queue of events to handle */
+static GAsyncQueue *events = NULL;
+static json_t exit_event;
+static void janus_gelfevh_event_free(json_t *event) {
+	if(!event || event == &exit_event)
+		return;
+	json_decref(event);
+}
+
+/* Gelf backend to send the events to */
+static char *backend = NULL;
+static char *port = NULL;
+static int sockfd;
+//static struct sockaddr_in servaddr;
+
+/* Parameter validation (for tweaking via Admin API) */
+static struct janus_json_parameter request_parameters[] = {
+	{"request", JSON_STRING, JANUS_JSON_PARAM_REQUIRED}
+};
+static struct janus_json_parameter tweak_parameters[] = {
+	{"events", JSON_STRING, 0},
+	{"backend", JSON_STRING, 0},
+	{"port", JSON_STRING, 0}
+};
+/* Error codes (for the tweaking via Admin API */
+#define JANUS_GELFEVH_ERROR_INVALID_REQUEST		411
+#define JANUS_GELFEVH_ERROR_MISSING_ELEMENT 	412
+#define JANUS_GELFEVH_ERROR_INVALID_ELEMENT 	413
+#define JANUS_GELFEVH_ERROR_UNKNOWN_ERROR 		499
+
+/* Plugin implementation */
+static void janus_gelfevh_connect(void) {
+	struct addrinfo *res = NULL;
+	janus_network_address addr;
+	janus_network_address_string_buffer addr_buf;
+	struct sockaddr_in servaddr;
+
+	if(getaddrinfo(backend, NULL, NULL, &res) != 0 ||
+		janus_network_address_from_sockaddr(res->ai_addr, &addr) != 0 ||
+		janus_network_address_to_string_buffer(&addr, &addr_buf) != 0) {
```

Comment:
The two lines above should have another tab of indentation, since they belong to the `if` above.

Reply:
This hasn't been resolved yet either.